src/main/java/net/dv8tion/jda/api/entities/Message.java

```diff
@@ -188,10 +203,20 @@ public interface Message extends ISnowflake, Formattable
     int MAX_STICKER_COUNT = 3;
 
     /**
-     * The maximum amount of {@link LayoutComponent LayoutComponents} that can be added to a message ({@value})
+     * The maximum amount of {@link MessageTopLevelComponent MessageTopLevelComponents} that can be added to a message's {@link #getComponents() root component list} when using the legacy component system.  ({@value})
      */
     int MAX_COMPONENT_COUNT = 5;
 
+    /**
+     * The maximum amount of {@link MessageTopLevelComponent MessageTopLevelComponents} that can be added to a message's {@link #getComponents() root component list} when using the {@link MessageRequest#useComponentsV2() V2 component system}.  ({@value})
```

Comment:
The [recent changelog](https://discord.com/developers/docs/change-log#raised-component-limits) said this limit was removed